app/models/framework_response.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class FrameworkResponse < VersionedModel
+  enum value_type: {
+    string: 'string',
+    array: 'array',
+    text: 'text',
+    json: 'json',
+  }
+
+  validates :value_type, presence: true, inclusion: { in: value_types }
+
+  belongs_to :framework_question
+  belongs_to :person_escort_record
+  has_many :dependents, class_name: 'FrameworkResponse',
+                        foreign_key: 'parent_id'
+
+  belongs_to :parent, class_name: 'FrameworkResponse', optional: true
+  has_and_belongs_to_many :flags, join_table: 'framework_responses_flags'
+
+  validates :value_text, absence: true, if: -> { json? || array? }
+  validates :value_json, absence: true, if: -> { string? || text? }
+
+  def value
```

Comment:
Only set/get value on a `framework_response`. The value is responsible where to set the actual answers based on their type in the database

Reply:
On second thought this will probably change when I implement validations as we will be defining different classes for each type of question so those might move specifically to the defined radio/textbox etc.